rust/datafusion/tests/sql.rs

```diff
@@ -1849,3 +1849,45 @@ async fn string_expressions() -> Result<()> {
     assert_eq!(expected, actual);
     Ok(())
 }
+
+#[tokio::test]
```

Comment:
Since there is also support in this PR for numeric types, I would also suggest some basic tests for IN lists with numbers as well (e.g. `c1 IN (1, 2 3)` as well as `c1 IN (1, NULL)`